src/prism.c

```diff
@@ -15379,6 +15379,15 @@ parse_expression_prefix(pm_parser_t *parser, pm_binding_power_t binding_power, b
             pm_parser_scope_pop(parser);
             pm_parser_current_param_name_restore(parser, saved_param_name);
 
+            /**
+             * If the final character is @. As is the case when defining
+             * methods to override the unary operators, we should ignore
+             * the @ in the same way we do for symbols.
+             */
+            if (name.type == PM_TOKEN_BANG && name.end[-1] == '@') {
```

Comment:
It looks like we're missing `PM_TOKEN_TILDE` from this check.

Can you reuse the code in `parse_operator_symbol`? Maybe by adding another function since this one is specific to creating symbols.

Something like:

```c
static const uint8_t *
parse_def_node_name(const pm_token_t *name) {
    switch (name->type) {
        case PM_TOKEN_TILDE:
        case PM_TOKEN_BANG:
            if (name->end[-1] == '@') return name->end - 1;
        /* fallthrough */
        default:
            return name->end;
    }
}
```

(Untested)